gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -140,11 +150,91 @@ def compute_inlier_mask(
 
         return inlier_idxs
 
+    def __get_priors_hilti_rig(
+        self,
+        i2Ti1_priors: Optional[Dict[Tuple[int, int], PosePrior]],
+        wRi_list: List[Optional[Rot3]],
+    ) -> gtsam.BinaryMeasurementsPoint3:
+        """Converts the priors from relative Pose3 priors to relative Point3 priors in world frame.
+        
+        If the priors are hard constraints (in the same rig), a hard-coded noise model is used. 
+        If the priors are soft constraints, the covariance from the PosePrior is used.
+
+        Soft constraints are only added between the 3rd rig cameras. 
+        Hard constraints are only added between the 3rd camera and other cameras in same rig.
+
+        Args: 
+            i2Ti1_priors: Relative pose priors between cameras, could be a hard or soft prior.
+            wRi_list: Absolute rotation estimates from Shonan averaging.
+        
+        Returns:
+            gtsam.BinaryMeasurementsPoint3 containing Point3 priors in world frame.
+        """
+        if i2Ti1_priors is None:
+            return gtsam.BinaryMeasurementsPoint3()
+
+        NUM_CAMERAS_IN_RIG = 5
+        BODY_FRAME_CAMERA = 2
+
+        def get_prior_in_world_frame(i2, i2Ti1_prior):
+            return wRi_list[i2].rotate(i2Ti1_prior.value).translation()
+
+        HARD_CONSTRAINT_NOISE_MODEL = gtsam.noiseModel.Constrained.All(1e4)
+        VALID_HARD_CONSTRAINT_EDGES = [(0, 2), (1, 2), (2, 3), (2, 4)]
+
+        w_i2ti1_priors = gtsam.BinaryMeasurementsPoint3()
+        priors_added = set()
+        for (i1, i2), i2Ti1_prior in i2Ti1_priors.items():
+            if i2Ti1_prior.type == PosePriorType.HARD_CONSTRAINT:
+                if (i1, i2) in VALID_HARD_CONSTRAINT_EDGES:
+                    w_i2ti1_priors.append(
+                        gtsam.BinaryMeasurementPoint3(
+                            i2,
+                            i1,
+                            get_prior_in_world_frame(i2, i2Ti1_prior),
+                            HARD_CONSTRAINT_NOISE_MODEL,
+                        )
+                    )
+                    priors_added.add((i2, i1))
+            else:
+                r1 = i1 // NUM_CAMERAS_IN_RIG
+                r2 = i2 // NUM_CAMERAS_IN_RIG
+                c1 = r1 * NUM_CAMERAS_IN_RIG + BODY_FRAME_CAMERA
+                c2 = r2 * NUM_CAMERAS_IN_RIG + BODY_FRAME_CAMERA
```

Comment:
Nit: I think you can do a combined check like how I do in rig shonan. But its down to personal preference I guess.